app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
```

Comment:
It's possible that I'm the one who misunderstood. I personally think that if a user has completed most of the signup process, and cancels IdV, we should take them to the recovery code page so they can complete that step. The only difference between LOA1 and LOA3 is that instead of seeing the recovery code page right after phone confirmation, you would normally see it after IdV. In the LOA 1 case, once you confirm your phone, you can no longer delete your account (the cancel link is not available on the recovery code page), so it seems logical to me that we wouldn't give the user the option to delete their account after phone confirmation in LOA 3.